### DISCUSSION TITLE
pkg installs do not respect `--appdir`

### DISCUSSION THREAD

**Discussion by setanarut** [Homebrew/discussions#5592] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f9bd2de89883ec9a59db3eab0f5475bce06d5bc
Last commit: 4 days ago
Core tap JSON: 05 Sep 21:56 UTC
Core cask tap JSON: 05 Sep 21:56 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--appdir=~/Applications", "--fontdir=~/Library/Fonts"]
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 12.7.6-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

`brew install google-earth-pro` command installs to wrong folder even though environment variable is home folder. It also asks for `Sudo` permission.

Command installs to `/Applications` instead `~/Applications`

`HOMEBREW_CASK_OPTS: ["--appdir=~/Applications", "--fontdir=~/Library/Fonts"]`

### Relevant casks

google-earth-pro

**Comment by SMillerDev** [comment, marked as answer]
It's not an app install but a pkg, so brew has no control over the location

**Reply by setanarut** [reply]
I am aware of this. The Homebrew package manager has to respect the environment variable option.

**Reply by SMillerDev** [reply]
The appdir setting defines where to move `.app` files when installing them. This is not a `.app` file so the setting has no effect

**Reply by setanarut** [reply]
> The appdir setting defines where to move `.app` files when installing them. This is not a `.app` file so the setting has no effect

But the last thing installed is an App

**Reply by setanarut** [reply]
> brew has no control over the location

Homebrew can install the pkg file to the desired location with a shell command;

URL -> https://dl.google.com/earth/client/advanced/current/GoogleEarthProMac-Intel.dmg

```Shell
installer -pkg /Volumes/Google\ Earth\ Pro\ 7.3.6.9796/Install\ Google\ Earth\ Pro\ 7.3.6.9796.pkg -target CurrentUserHomeDirectory
```

**Reply by SMillerDev** [reply]
> Homebrew can install the pkg file to the desired location with a shell command

Homebrew can try, but it has no guarantees if that works because it's up to the pkg developer to support that.

**Reply by setanarut** [reply]
> > Homebrew can install the pkg file to the desired location with a shell command
> 
> Homebrew can try, but it has no guarantees if that works because it's up to the pkg developer to support that.

This is a testable condition so it can be guaranteed.

**Reply by SMillerDev** [reply]
If you're confident it can be done, please make a pull request to support it.

**Comment by setanarut** [comment]
@SMillerDev If you do not have any idea or solutions, please do not reply.

**Reply by p-linnane** [reply]
@SMillerDev is a long-term Homebrew maintainer who is offering a reason why this is not working like you want. You don't have to like what you hear, but that's how it is. Please contribute a PR to support this and we are happy to review it. That is much more helpful than reacting to maintainers with :-1:.

**Reply by setanarut** [reply]
Why do I care who he is? I reported a bug and a problem I had. You can say, "We're sorry, we can't fix this."

**Reply by carlocab** [reply]
> Why do I care who he is?

Because you should probably be nicer to a person who uses their spare time to work on a tool that's given away for free, without asking for anything in return.

That just seems like common decency to me. Of course, if you don't believe in common decency: please take your complaints elsewhere.

**Reply by setanarut** [reply]
> > Why do I care who he is?
> 
> Because you should probably be nicer to a person who uses their spare time to work on a tool that's given away for free, without asking for anything in return.
> 
> That just seems like common decency to me. Of course, if you don't believe in common decency: please take your complaints elsewhere.

I also wasted my time by reporting bugs. I am not grateful to anyone for my bad software experience.

**Reply by carlocab** [reply]
> I also wasted my time by reporting bugs.

The Homebrew project will live without your bug reports. OTOH, it won't if people like @SMillerDev don't spend their time working on it. These two things are not the same.

In any case: being willing to take the time to report bugs doesn't exclude anyone from common decency.

> I am not grateful to anyone for my bad software experience.

If your experience of using `brew`, all things considered, is bad, then please don't use it. If that's not your overall experience: then this comment is not relevant.

**Reply by setanarut** [reply]
> OTOH, it won't if people like @SMillerDev don't spend their time working on it

He already doesn't spend his time.
